### PR TITLE
Reject negative process grid dimensions

### PR DESCRIPTION
**Pull request by fallintoplace**
## Summary

Reject negative `pdims` values during grid descriptor validation before checking the process-grid product.

Previously, `checkConfig` only used product semantics for explicit process grids, so a configuration such as `[-2, -2]` could pass on four ranks. The later process-grid indexing code assumes positive dimensions for division, modulo, communicator setup, and pencil decomposition.

This also widens the product calculation to `int64_t` so validation does not overflow before comparing with the rank count.

## Tests

- `git diff --check`
- Added API regression coverage for negative `pdims` pairs whose product equals the test rank count

Local build/test execution was not available in this checkout because `clang-format`, `nvcc`, and `mpicxx` are not installed.

### CODE REVIEW

File: tests/ctest/api_tests.cc [hunk 1]
```diff
@@ -863,6 +863,16 @@ TEST_F(ApiGridDescCreateTest, RejectsInvalidConfigs) {
   config.pdims[1] = 1;
   expectGridDescCreateInvalid(config);
 
+  config = distributedConfig();
+  config.pdims[0] = -2;
+  config.pdims[1] = -2;
+  expectGridDescCreateInvalid(config);
+
+  config = distributedConfig();
```
Comment (romerojosh) [inline tests/ctest/api_tests.cc]: This second test is redundant so it can be removed.

**Review by romerojosh — COMMENTED** [review]
Thanks for the contribution @fallintoplace! I've left a comment to address but otherwise this looks good. Definitely covering a rare edge case, but better to catch this user error than not.

File: tests/ctest/api_tests.cc (hunk 1 above)
Comment (romerojosh) [reply tests/ctest/api_tests.cc]: Went ahead and removed the redundant test. Will merge when the build workflow passes. Thanks for the contribution!

Comment (fallintoplace) [reply tests/ctest/api_tests.cc]: Thank you for your support. I apologize for not seeing this comment here.
